plugin-parser/mlops-plugin-parser.cc

```diff
@@ -338,11 +338,20 @@ PKGMGR_MDPARSER_PLUGIN_INSTALL (const char *pkgid, const char *appid, GList *met
 
   char *res_type = NULL;
   ret = pkgmgrinfo_pkginfo_get_res_type (handle, &res_type);
+  if (ret != PMINFO_R_OK) {
+    _E ("Failed to get res type.");
+    pkgmgrinfo_pkginfo_destroy_pkginfo (handle);
+    return -1;
```

Comment:
How about return `ret` instead of -1?

`pkgmgrinfo_pkginfo_get_res_type` pkgmgr-info pacakge and can returns various status.
```
API int pkgmgrinfo_pkginfo_get_res_type(pkgmgrinfo_pkginfo_h handle,
                char **res_type)
{
         pkgmgr_pkginfo_x *info = (pkgmgr_pkginfo_x *)handle;
 
         retvm_if(handle == NULL, PMINFO_R_EINVAL, "pkginfo handle is NULL\n");
         retvm_if(res_type == NULL, PMINFO_R_EINVAL,
                         "Argument supplied to hold return value is NULL\n");
 
         if (info->pkg_info == NULL)
                 return PMINFO_R_ERROR;
 
         if (info->pkg_info->res_type == NULL)
                 return PMINFO_R_ENOENT;
 
         *res_type = (char *)info->pkg_info->res_type;
 
         return PMINFO_R_OK;
}
```
[source code](https://review.tizen.org/git/?p=platform/core/appfw/pkgmgr-info.git;a=blob;f=src/pkgmgrinfo_pkginfo.c;h=b72c63d6fdf40c78e4d91156d33500d443a8ff4a;hb=HEAD#l887)

There are several enum value for this package
```
enum {
        PMINFO_R_ENOENT = -3,           /**< No result */
        PMINFO_R_EINVAL = -2,           /**< Invalid argument */
        PMINFO_R_ERROR = -1,            /**< General error */
        PMINFO_R_OK = 0                 /**< General success */
};
```
[source code](https://review.tizen.org/git/?p=platform/core/appfw/pkgmgr-info.git;a=blob;f=include/pkgmgrinfo_type.h;h=7f83f1645897fc17c39800cb2017230a8777dfb4;hb=HEAD#l124)

I think returning `ret` instead of -1 would make debugging easier if there's a problem.

Reply:
There is no Tizen's guide on what plugin-parser should return, but it would be great to check return values like you suggested in terms of code readability and TDD manner.

By the way, `PMINFO_R_*` enum values are for the pkgmgr-info itself, not for the plugin-parser. The plugin-parser may need its own return values which only used in unit tests or just check 0 or not as @jaeyun-jung [mentioned](https://github.com/nnstreamer/deviceMLOps.MLAgent/pull/38#discussion_r1542300060).

I'll add more unit tests to cover more error cases, and update the plugin-parser accordingly, Thanks!